src/main/java/com/conveyal/datatools/manager/models/FeedSource.java

```diff
@@ -459,108 +459,73 @@ public String latestVersionId() {
-        if (deployedFeedVersion == null) {
-            Project project = Persistence.projects.getById(this.projectId);
-            if (project.pinnedDeploymentId != null) {
-                Deployment deployment = Persistence.deployments.getByIdLimitedFields(
-                    project.pinnedDeploymentId,
-                    "pinnedfeedVersionIds", "feedVersionIds"
-                );
-                FeedVersion feedVersion = getLatestDeployedFeedVersionForFeedSource(deployment.pinnedfeedVersionIds);
-                if (feedVersion != null) {
-                    // This feed version will be the latest pinned version for this feed source, if available.
-                    deployedFeedVersion = feedVersion;
-                } else {
-                    feedVersion = getLatestDeployedFeedVersionForFeedSource(deployment.feedVersionIds);
-                    if (feedVersion != null) {
-                        // This feed version will be the latest version for this feed source, if available.
-                        deployedFeedVersion = feedVersion;
-                    }
-                }
-            }
-
-            // If there is no pinned deployment or none of the deployment's feed versions are related to this feed source,
-            // find the latest feed version for this feed source.
-            if (deployedFeedVersion == null) {
-                // Get all deployments for this project.
-                List<Deployment> deployments = Persistence.deployments.getFilteredLimitedFields(
-                    eq("projectId", this.projectId),
-                    Sorts.descending("lastUpdated"),
-                    "feedVersionIds"
-                );
-                // Iterate through deployments newest to oldest.
-                for (Deployment deployment : deployments) {
-                    FeedVersion feedVersion = getLatestDeployedFeedVersionForFeedSource(deployment.feedVersionIds);
-                    if (feedVersion != null) {
-                        // This feed version will be the latest feed version for this feed source from the newest
-                        // deployment.
-                        deployedFeedVersion = feedVersion;
-                        break;
-                    }
-                }
-            }
+     * If a project has a "pinned" deployment, return the feed version from this pinned deployment. If it is not
+     * available return null and don't attempt to get the feed version from the latest deployment.
+     *
+     * If a project does not have a "pinned" deployment, return the latest deployment's feed versions for this feed
+     * source, if available.
+     */
+    public FeedVersionDeployed retrieveDeployedFeedVersion() {
+        if (deployedFeedVersionDefined) {
+            return deployedFeedVersion;
+        }
+        Project project = Persistence.projects.getById(projectId);
+        if (project.pinnedDeploymentId != null && !project.pinnedDeploymentId.isEmpty()) {
```

Comment:
I suspect this call is what's slowing down things significantly. What do we think about making this call optional? @philip-cline

Reply:
I would liked to have included this in the Mongo query, but hit my limitations with Mongo. If this is an issue I can look at this again and attempt to include?

Reply:
I agree this can be a separate PR!